src/tool/main.rs

```diff
@@ -1,27 +1,50 @@
 // Copyright 2019-2023 ChainSafe Systems
 // SPDX-License-Identifier: Apache-2.0, MIT
 
+use crate::cli_shared::cli::*;
+
 use std::ffi::OsString;
 
 use super::subcommands::Cli;
 use clap::Parser;
 
+use crate::utils::{io::read_file_to_string, io::read_toml};
+
 use super::subcommands::Subcommand;
 
+fn read_config() -> anyhow::Result<Config> {
+    let opts = CliOpts::default();
+    let path = find_config_path(&opts);
```

Comment:
There are no `CliOpts` here.

Reply:
Indeed. I can create a small util `find_config_path` that doesn't require `opts`, sounds ok?

Reply:
How can you find non-default configuration paths without taking any arguments?

Reply:
I know, this is missing for now but will handle that case.